lib/routes/devices-and-sessions.js

```diff
@@ -237,6 +244,99 @@ module.exports = (log, db, config, customs, push, devices) => {
         }
       }
     },
+    {
+      method: 'GET',
+      path: '/account/device/messages',
+      config: {
+        validate: {
+          query: {
+            index: isA.string().alphanum().optional(),
+            limit: isA.number().optional().min(0).max(100),
+          }
+        },
+        auth: {
+          strategy: 'sessionToken'
+        },
+        response: {
+          schema: PUSHBOX_RETRIEVE_SCHEMA
+        }
+      },
+      handler (request, reply) {
+        log.begin('Account.deviceMessages', request)
+
+        const sessionToken = request.auth.credentials
+        const uid = sessionToken.uid
+        const deviceId = sessionToken.deviceId
+        const ip = request.app.clientAddress
+        const query = request.query || {}
+        const {index} = query
+        const limit = query.limit !== undefined ? Math.min(100, query.limit) : 100
+
+        return customs.checkAuthenticated('deviceMessages', ip, uid)
+          .then(() => oAuthMint.getOAuthToken(sessionToken, PUSHBOX_OAUTH_SCOPE_RECV))
+          .then(resp => {
+            const {access_token: oAuthToken} = resp
+            return pushbox.retrieve(uid, deviceId, oAuthToken, limit, index)
+          })
+          .then(reply, reply)
+      }
+    },
+    {
+      method: 'POST',
+      path: '/account/devices/messages',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            to: isA.string().required(),
```

Comment:
Is there any additional restrictions we could put here? Maybe max length or character types?